moveit_core/collision_detection/include/moveit/collision_detection/world_diff.h

```diff
@@ -39,7 +39,6 @@
 
 #include <moveit/collision_detection/world.h>
 #include <moveit/macros/class_forward.h>
-#include <boost/weak_ptr.hpp>
```

Comment:
Well, you can't make `weak_ptrs` without `shared_ptrs`. Since we have `std::shared_ptr` typedefs available from a macro in `#include <moveit/macros/declare_ptr.h>`, that will have pulled in `#include <memory>`.

But I am a fan of the rule "if you use a symbol from a header, include it", so I'll add it.